### DISCUSSION TITLE
How can we contribute to Docs?

### DISCUSSION THREAD

**Discussion by init27** [Nixtla/neuralforecast#173] [General]
Hello Hello! 

I found your work via R/ML. Thanks for creating the framework! 

I'm not a Time Series expert and neither do I use it at work. However, I noticed that the documentation is quite bare bones at the moment. 

I would love to contribute and was curious how can we get started or what might be the best first places to contribute?

Thanks in advance!

**Comment by AzulGarza** [comment]
Hi @init27!  👋 

We use `nbdev` to develop `neuralforecast`. You can find the documentation [here](https://nbdev.fast.ai/). 

Broadly speaking:
- Once you have made a git clone of the repo, the first thing you need to do is create a conda environment with the `environment.yml` file.
- After that, you have to activate the environment and install the library interactively using `pip install -e .`, and run `nbdev_install_git_hooks`. 
- From here, you can modify the documentation and the library. You have to change the notebooks inside the nbs folder (revising the cells in markdown changes to documentation). 
- Finally, it remains to reflect the changes in the python modules and html files. For that, just run `nbdev_build_lib` (changes in modules) and `nbdev_build_docs` (changes in the documentation). Before doing push, it is also good practice to run `nbdev_clean_nbs` (to clean notebooks cache), `nbdev_diff_nbs` (to verify that there are no differences between notebooks and scripts), and finally `nbdev_test_nbs` to run the notebooks and verify that everything runs fine. (All these steps can be executed using `make nbdev_test_flow`).
- You can see changes in the documentation locally by running `make docs_serve`.

Currently, the only file we do not accept modifications to is `index.ipynb`. :)

For a more detailed explanation you can see our [contributing guidelines](https://github.com/Nixtla/neuralforecast/blob/main/CONTRIBUTING.md). 📝 

We are excited to see your contributions! 🎉 🔥 
Please let me know if you have more questions. :)
